src/auto/AutoCriterion.lua

```diff
@@ -0,0 +1,26 @@
+local AutoCriterion, parent = torch.class('nn.AutoCriterion', 'nn.Criterion')
```

Comment:
So the issue is that it would require `autograd` to be a global package, because of the funky design of `torch.class`. I'd rather keep it local.